File: backend/app/schemas/ranking.py
from pydantic import BaseModel
from datetime import datetime
from uuid import UUID


class RankingEntry(BaseModel):
    rank: int
    pet_id: UUID
    pet_name: str
    pet_photo_url: str | None = None
    owner_username: str
    likes_this_week: int
    updated_at: datetime

    class Config:
        from_attributes = True

File: backend-node/src/schemas/ranking.ts
import { z } from "zod";

export const RankingEntry = z.object({
  rank: z.number().int(),
  pet_id: z.string().uuid(),
  pet_name: z.string(),
  pet_photo_url: z.string().url().nullable().optional(),
  owner_username: z.string(),
  likes_this_week: z.number().int(),
  updated_at: z.string(),
});

export const RankingResponse = z.object({
  items: z.array(RankingEntry),
  updated_at: z.string(),
});

export type RankingEntry = z.infer<typeof RankingEntry>;
export type RankingResponse = z.infer<typeof RankingResponse>;
